packages/ui/src/lib/imageDownloadButton/ImageDownloadButton.svelte

```diff
@@ -62,107 +51,49 @@
+			const area = bbox.width * bbox.height;
 
-		const svgString = new XMLSerializer().serializeToString(svgNodeClone);
-		const svgURL = 'data:image/svg+xml;base64,' + stringToBase64(svgString);
-		// the non base-64 encoded URL would be: "data:image/svg+xml;utf8," + svgString;
-
-		// an alternative is:
-		// const svgURL = URL.createObjectURL(new Blob([svgNodeClone.outerHTML],{type:'image/svg+xml;charset=utf-8'}))
-
-		const img = await createHTMLImageFromURL(svgURL);
-
-		if (format === 'SVG') {
-			downloadFromURL(svgURL);
-		} else {
-			img.src = svgURL;
-
-			const w = svgNode!.clientWidth * scaleFactor;
-			const h = svgNode!.clientHeight * scaleFactor;
-
-			const canvas = new OffscreenCanvas(w, h);
-
-			canvas.getContext('2d')?.drawImage(img, 0, 0, w, h);
+			if (area > largestArea) {
+				largestSVG = element;
+				largestArea = area;
+			}
+		});
 
-			canvas.convertToBlob().then((blob) => {
-				downloadFromURL(URL.createObjectURL(blob));
-			});
-		}
+		return largestSVG;
 	};
 
-	///
-
-	const preserveDrawingBuffer = () => {
-		function wrapGetContext(ContextClass: typeof HTMLCanvasElement | typeof OffscreenCanvas) {
-			const isWebGL = /webgl/i;
-
-			ContextClass.prototype.getContext = (function (origFn: (type: any, attributes?: any) => any) {
-				return function (this: HTMLCanvasElement | OffscreenCanvas, type: any, attributes: any) {
-					if (isWebGL.test(type)) {
-						attributes = Object.assign({}, attributes || {}, { preserveDrawingBuffer: true });
-					}
-					return origFn.call(this, type, attributes);
-				};
-			})(ContextClass.prototype.getContext);
-		}
-
-		if (typeof HTMLCanvasElement !== 'undefined') {
-			wrapGetContext(HTMLCanvasElement);
-		}
-		if (typeof OffscreenCanvas !== 'undefined') {
-			wrapGetContext(OffscreenCanvas);
+	// Filter elemts to hide based on class
+	const filter = (node: HTMLElement) => {
+		if (node instanceof HTMLElement) {
+			const ignoreAttributes = ['data-capture-ignore'];
+			return !ignoreAttributes.some((attribute) => node.hasAttribute(attribute));
 		}
-	};
-
-	const downloadFromHTML = async (el: HTMLElement) => {
-		preserveDrawingBuffer();
-
-		const canvas = await html2canvas(el, {
-			onclone: (clone: Document) => {
-				if (idToPad) {
-					clone.getElementById(idToPad)!.style.padding = padding;
-				}
-			},
-			windowWidth: 1500
-		});
-		const image = canvas.toDataURL('image/png', 1.0);
-		downloadFromURL(image);
+		return node;
 	};
 
 	const download = async () => {
-		if (svgNode) {
-			downloadFromSVG();
-		} else if (htmlNode) {
-			downloadFromHTML(htmlNode);
+		const captureOptions = { style: { padding: padding }, filter };
+		if (format === 'SVG') {
+			const svgNode = findNearestChildSvg(htmlNode);
+			svgNode !== null
+				? toSvg(svgNode, captureOptions).then((dataUrl) => downloadFromURL(dataUrl))
+				: console.warn('No svgNode found');
+		} else if (format === 'PNG') {
+			toPng(htmlNode, captureOptions).then((dataUrl) => downloadFromURL(dataUrl));
```

Comment:
I think you can do something like `{...captureOptions, pixelRatio}` (assuming you also do `export let pixelRatio = 1;`) to allow control of the image resolution.

https://www.npmjs.com/package/html-to-image